spec/controllers/providers/allocations_controller_spec.rb

```diff
@@ -0,0 +1,124 @@
+require "rails_helper"
+
+RSpec.describe Providers::AllocationsController, type: :controller do
```

Comment:
Aren't we duplicating a lot testing here? We have a feature spec which we prefer over these.

Reply:
No, they test all the low hanging fruits that is not tested in the feature spec.

Such as which partial template is expected to be rendered, and the training providers that are all expected to be.